### PR TITLE
Rename `State` to `KeyValueStore`

### LINKED ISSUE #2528: Rename `State` to `Storage` or `GlobalStorage` or something
See https://github.com/cursorless-dev/cursorless/blob/main/packages/common/src/ide/types/State.ts

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #2528

## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

**Review by pokey — COMMENTED** [review]
This looks fine, tho fwiw in VSCode they use `storage` for a place to put files. Maybe we should use `KeyValueStore`?

**Comment by AndreasArvidsson** [comment]
> This looks fine, tho fwiw in VSCode they use `storage` for a place to put files. Maybe we should use `KeyValueStore`?

We could or we could just ignore what vscode calls it? :D

**Comment by pokey** [comment]
Sure but I think storage kind of implies what vscode is doing. Is KeyValueStore bad? I think it's pretty clear

**Comment by AndreasArvidsson** [comment]
Renamed to `KeyValueStore`

File: packages/common/src/ide/types/KeyValueStore.ts [hunk 1]
```diff
@@ -9,38 +9,40 @@ export type TutorialProgress = Partial<
...
 }
-export type StateKey = keyof StateData;
+export type KeyValueStoreKey = keyof KeyValueStoreData;
 
 /**
- * A mapping from allowable state keys to their default values
+ * A mapping from allowable storage keys to their default values
  */
-export const STATE_DEFAULTS: StateData = {
+export const STORAGE_DEFAULTS: KeyValueStoreData = {
```
Comment (pokey) [inline packages/common/src/ide/types/KeyValueStore.ts:21]:
```suggestion
export const KEY_VALUE_STORE_DEFAULTS: KeyValueStoreData = {
```

File: packages/cursorless-tutorial/src/loadTutorial.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export async function loadTutorial(
   tutorialId: TutorialId,
   customSpokenFormGenerator: CustomSpokenFormGenerator,
   rawContent: RawTutorialContent,
-  globalState: State,
+  storage: KeyValueStore,
```
Comment (pokey) [inline packages/cursorless-tutorial/src/loadTutorial.ts:18]:
```suggestion
  keyValueStore: KeyValueStore,
```

File: packages/common/src/ide/types/KeyValueStore.ts [hunk 3]
```diff
@@ -9,38 +9,40 @@ export type TutorialProgress = Partial<
...
+export interface KeyValueStoreData {
   hideInferenceWarning: boolean;
   tutorialProgress: TutorialProgress;
 }
-export type StateKey = keyof StateData;
+export type KeyValueStoreKey = keyof KeyValueStoreData;
 
 /**
- * A mapping from allowable state keys to their default values
+ * A mapping from allowable storage keys to their default values
```
Comment (pokey) [inline packages/common/src/ide/types/KeyValueStore.ts:19]:
```suggestion
 * A mapping from allowable key value store keys to their default values
```

**Review by pokey — COMMENTED** [review]
looks good with minor tweaks

**Comment by AndreasArvidsson** [comment]
Changes accepted

**Review by pokey — APPROVED** [review]

**Pull request merged**